### ISSUE TITLE
Do not fetch quotes of former holdings

### ISSUE THREAD

**Issue by dtslvr** [ghostfolio/ghostfolio#2531]
### Problem

There is an inefficiency that [quotes](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/services/data-provider/data-provider.service.ts#L237) of all holdings are loaded, even if they are no longer owned at all.

### Solution

Quotes should not be loaded for former holdings (`quantity` equals 0).

### Steps to reproduce

1. Create an account
1. Import activities ([ghostfolio-export-AAPL-202310220844.json](https://github.com/ghostfolio/ghostfolio/files/13062540/ghostfolio-export-AAPL-202310220844.json))
1. Open `http://localhost:4200/en/portfolio/activities?dataSource=YAHOO&symbol=AAPL&positionDetailDialog=true`
1. See `DEBUG Fetched 1 quote from YAHOO in 0.134 seconds` in server log (should be `Fetched 0 quote in 0.001 seconds`)

**Comment by Dhoni77** [comment]
Hi @dtslvr  I would like to take this.

**Comment by Dhoni77** [comment]
Hi @dtslvr 
I have a small doubt.
In the activities provided the quantity is 1 for both activities.
So in the activity table if an activity has quanity < 1,  then when we click on the activity the corresponding quote should not be fetched for that activity.
Am  I correct here ?

**Comment by dtslvr** [comment]
Hi @Dhoni77

The example of the `json` export is designed like this: The two activities are one of type `BUY` (quantity = 1) and one of type `SELL` (quantity = 1) which leads to a net quantity of 0 (`+ 1 - 1 = 0`). Therefore, quotes should not be fetched because the user does not own this asset anymore. Okay?

**Comment by Dhoni77** [comment]
Hi @dtslvr
For example we have 3 activities

The symbols are same.

1st Activity has A account and has type as Buy and quantity 1
2nd Activity has A account and has type as Sell and quantity 1
3rd Activity has B account and has type as Buy and quantity 1

So now should I take the B account into consideration for calculating net quantity or only the same A accounts.

**Comment by dtslvr** [comment]
> So now should I take the B account into consideration for calculating net quantity or only the same A accounts.

I don't think you need to calculate this by yourself. `getDetails()` of `PortfolioService` already provides the (aggregated) `quantity` (see `holdings`) per symbol across different accounts.

**Comment by dtslvr** [comment]
Superseded by https://github.com/ghostfolio/ghostfolio/issues/5395.
